src/java/org/jivesoftware/openfire/roster/Roster.java

```diff
@@ -625,7 +625,7 @@ public void broadcastPresence(Presence packet) {
             }
         }
         // Broadcast presence to shared contacts whose subscription status is FROM
-        for (String contact : implicitFrom.keySet()) {
+        for (String contact : ((Map<String,Set<String>>) implicitFrom).keySet()) { // Cast prevents Java compatibility issue as described in OF-1116. Remove the cast when Java 7 support is dropped from Openfire.

```

Comment:
In Java 8, I don't see a difference anyway, because all the `ConcurrentMap` methods are default methods on `Map` now.